src/app/inventory/store/stats.ts

```diff
@@ -426,15 +439,40 @@ function applyPlugsToStats(
     compareBy((s) => s.plugOptions.length)
   );
   for (const socket of sortedSockets) {
-    attachPlugStats(socket, existingStatsByHash, statDisplaysByStatHash);
+    attachPlugStats(createdItem, socket, existingStatsByHash, statDisplaysByStatHash);
   }
 }
 
+/**
+ * Adept raid weapons that were randomly acquired can be enhanced to get an enhanced intrinsic,
+ * at which point they're functionally crafted.
+ * Their intrinsic says "conditionally +2 to some stat", but they get +3 because that's how
+ * masterworked adepts behave, and an additional +1 by reaching weapon level 20. There's no
+ * basis for this behavior in the defs, so we cheat when we calculate live stats and attribute
+ * these stats to the intrinsic since that's the "masterwork".
+ */
+function getPlugStatValue(
+  createdItem: DimItem,
+  plug: PluggableInventoryItemDefinition,
+  stat: DestinyItemInvestmentStatDefinition
+) {
+  if (
+    stat.isConditionallyActive &&
+    enhancedIntrinsics.has(plug.hash) &&
+    adeptWeaponHashes.includes(createdItem.hash)
+  ) {
+    return stat.value * ((createdItem.craftedInfo?.level ?? 0) >= 20 ? 2 : 1.5);
```

Comment:
this feels like a really complicated way, based on a coincidence of numbers, to effectively say
`+ (level >= 20 ? 1 : 2)` or whatever the right numbers would be.
any reason you're going by ratio/multiplication here? does it apply to some/any other case?